### ISSUE TITLE
💅 False positive by lint/nursery/noUselessUndefined

### ISSUE THREAD

**Issue by atheck** [biomejs/biome#6577]
### Environment information

```bash
CLI:
  Version:                      2.0.5
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           windows

Environment:
  BIOME_LOG_PATH:               unset
  BIOME_LOG_PREFIX_NAME:        unset
  BIOME_CONFIG_PATH:            unset
  BIOME_THREADS:                unset
  NO_COLOR:                     unset
  TERM:                         unset
  JS_RUNTIME_VERSION:           v22.14.0
  JS_RUNTIME_NAME:              node
  NODE_PACKAGE_MANAGER:         npm/10.9.2

Biome Configuration:
  Status:                       Loaded successfully
  Path:                         biome.jsonc
  Formatter enabled:            true
  Linter enabled:               true
  Assist enabled:               true
  VCS enabled:                  true

Linter:
  JavaScript enabled:           unset
  JSON enabled:                 unset
  CSS enabled:                  unset
  GraphQL enabled:              unset
  Recommended:                  unset
  Enabled rules:
    a11y/noAccessKey
    a11y/noAriaHiddenOnFocusable
    a11y/noAriaUnsupportedElements
    a11y/noAutofocus
    a11y/noDistractingElements
    a11y/noHeaderScope
    a11y/noInteractiveElementToNoninteractiveRole
    a11y/noLabelWithoutControl
    a11y/noNoninteractiveElementToInteractiveRole
    a11y/noNoninteractiveTabindex
    a11y/noPositiveTabindex
    a11y/noRedundantAlt
    a11y/noRedundantRoles
    a11y/noStaticElementInteractions
    a11y/noSvgWithoutTitle
    a11y/useAltText
    a11y/useAnchorContent
    a11y/useAriaActivedescendantWithTabindex
    a11y/useAriaPropsForRole
    a11y/useAriaPropsSupportedByRole
    a11y/useButtonType
    a11y/useFocusableInteractive
    a11y/useGenericFontNames
    a11y/useHeadingContent
    a11y/useHtmlLang
    a11y/useIframeTitle
    a11y/useKeyWithClickEvents
    a11y/useKeyWithMouseEvents
    a11y/useMediaCaption
    a11y/useSemanticElements
    a11y/useValidAnchor
    a11y/useValidAriaProps
    a11y/useValidAriaRole
    a11y/useValidAriaValues
    a11y/useValidAutocomplete
    a11y/useValidLang
    complexity/noAdjacentSpacesInRegex
    complexity/noArguments
    complexity/noBannedTypes
    complexity/noCommaOperator
    complexity/noEmptyTypeParameters
    complexity/noExcessiveNestedTestSuites
    complexity/noExtraBooleanCast
    complexity/noFlatMapIdentity
    complexity/noStaticOnlyClass
    complexity/noThisInStatic
    complexity/noUselessCatch
    complexity/noUselessConstructor
    complexity/noUselessContinue
    complexity/noUselessEmptyExport
    complexity/noUselessEscapeInRegex
    complexity/noUselessFragments
    complexity/noUselessLabel
    complexity/noUselessLoneBlockStatements
    complexity/noUselessRename
    complexity/noUselessStringConcat
    complexity/noUselessStringRaw
    complexity/noUselessSwitchCase
    complexity/noUselessTernary
    complexity/noUselessThisAlias
    complexity/noUselessTypeConstraint
    complexity/noUselessUndefinedInitialization
    complexity/noVoid
    complexity/useArrowFunction
    complexity/useDateNow
    complexity/useFlatMap
    complexity/useLiteralKeys
    complexity/useNumericLiterals
    complexity/useOptionalChain
    complexity/useRegexLiterals
    complexity/useSimpleNumberKeys
    complexity/useWhile
    correctness/noChildrenProp
    correctness/noConstAssign
    correctness/noConstantCondition
    correctness/noConstantMathMinMaxClamp
    correctness/noConstructorReturn
    correctness/noEmptyCharacterClassInRegex
    correctness/noEmptyPattern
    correctness/noGlobalObjectCalls
    correctness/noInnerDeclarations
    correctness/noInvalidBuiltinInstantiation
    correctness/noInvalidConstructorSuper
    correctness/noInvalidDirectionInLinearGradient
    correctness/noInvalidGridAreas
    correctness/noInvalidPositionAtImportRule
    correctness/noMissingVarFunction
    correctness/noNonoctalDecimalEscape
    correctness/noPrecisionLoss
    correctness/noRenderReturnValue
    correctness/noSelfAssign
    correctness/noSetterReturn
    correctness/noStringCaseMismatch
    correctness/noSwitchDeclarations
    correctness/noUnknownFunction
    correctness/noUnknownMediaFeatureName
    correctness/noUnknownProperty
    correctness/noUnknownPseudoClass
    correctness/noUnknownPseudoElement
    correctness/noUnknownTypeSelector
    correctness/noUnknownUnit
    correctness/noUnmatchableAnbSelector
    correctness/noUnreachable
    correctness/noUnreachableSuper
    correctness/noUnsafeFinally
    correctness/noUnsafeOptionalChaining
    correctness/noUnusedFunctionParameters
    correctness/noUnusedImports
    correctness/noUnusedLabels
    correctness/noUnusedPrivateClassMembers
    correctness/noUnusedVariables
    correctness/noVoidElementsWithChildren
    correctness/noVoidTypeReturn
    correctness/useExhaustiveDependencies
    correctness/useHookAtTopLevel
    correctness/useIsNan
    correctness/useJsxKeyInIterable
    correctness/useValidForDirection
    correctness/useValidTypeof
    correctness/useYield
    nursery/noAwaitInLoop
    nursery/noBitwiseOperators
    nursery/noConstantBinaryExpression
    nursery/noFloatingPromises
    nursery/noGlobalDirnameFilename
    nursery/noImportCycles
    nursery/noImportantStyles
    nursery/noNestedComponentDefinitions
    nursery/noNoninteractiveElementInteractions
    nursery/noProcessGlobal
    nursery/noReactPropAssign
    nursery/noTsIgnore
    nursery/noUnknownAtRule
    nursery/noUselessBackrefInRegex
    nursery/noUselessEscapeInString
    nursery/noUselessUndefined
    nursery/useAdjacentGetterSetter
    nursery/useConsistentObjectDefinition
    nursery/useConsistentResponse
    nursery/useExhaustiveSwitchCases
    nursery/useExportsLast
    nursery/useGoogleFontPreconnect
    nursery/useIndexOf
    nursery/useJsonImportAttribute
    nursery/useNumericSeparators
    nursery/useObjectSpread
    nursery/useParseIntRadix
    nursery/useSingleJsDocAsterisk
    nursery/useSymbolDescription
    nursery/useUniqueElementIds
    performance/noAccumulatingSpread
    performance/noBarrelFile
    performance/noDelete
    performance/noDynamicNamespaceImportAccess
    performance/noNamespaceImport
    performance/noReExportAll
    performance/useTopLevelRegex
    security/noBlankTarget
    security/noDangerouslySetInnerHtml
    security/noDangerouslySetInnerHtmlWithChildren
    security/noGlobalEval
    style/noCommonJs
    style/noDefaultExport
    style/noDescendingSpecificity
    style/noDoneCallback
    style/noInferrableTypes
    style/noNamespace
    style/noNegationElse
    style/noNestedTernary
    style/noNonNullAssertion
    style/noParameterAssign
    style/noProcessEnv
    style/noShoutyConstants
    style/noSubstr
    style/noUnusedTemplateLiteral
    style/noUselessElse
    style/noValueAtRule
    style/noYodaExpression
    style/useArrayLiterals
    style/useAsConstAssertion
    style/useAtIndex
    style/useBlockStatements
    style/useCollapsedElseIf
    style/useCollapsedIf
    style/useConsistentArrayType
    style/useConsistentBuiltinInstantiation
    style/useConsistentMemberAccessibility
    style/useConst
    style/useDefaultParameterLast
    style/useDeprecatedReason
    style/useEnumInitializers
    style/useExplicitLengthCheck
    style/useExponentiationOperator
    style/useExportType
    style/useForOf
    style/useImportType
    style/useLiteralEnumMembers
    style/useNamingConvention
    style/useNodeAssertStrict
    style/useNumberNamespace
    style/useSelfClosingElements
    style/useShorthandAssign
    style/useShorthandFunctionType
    style/useSingleVarDeclarator
    style/useTemplate
    style/useThrowNewError
    style/useThrowOnlyError
    style/useTrimStartEnd
    suspicious/noApproximativeNumericConstant
    suspicious/noArrayIndexKey
    suspicious/noAssignInExpressions
    suspicious/noAsyncPromiseExecutor
    suspicious/noCatchAssign
    suspicious/noClassAssign
    suspicious/noCommentText
    suspicious/noCompareNegZero
    suspicious/noConfusingLabels
    suspicious/noConfusingVoidType
    suspicious/noConsole
    suspicious/noConstEnum
    suspicious/noControlCharactersInRegex
    suspicious/noDebugger
    suspicious/noDocumentCookie
    suspicious/noDoubleEquals
    suspicious/noDuplicateAtImportRules
    suspicious/noDuplicateCase
    suspicious/noDuplicateClassMembers
    suspicious/noDuplicateCustomProperties
    suspicious/noDuplicateElseIf
    suspicious/noDuplicateFields
    suspicious/noDuplicateFontNames
    suspicious/noDuplicateJsxProps
    suspicious/noDuplicateObjectKeys
    suspicious/noDuplicateParameters
    suspicious/noDuplicateProperties
    suspicious/noDuplicateSelectorsKeyframeBlock
    suspicious/noDuplicateTestHooks
    suspicious/noEmptyBlock
    suspicious/noEmptyBlockStatements
    suspicious/noEvolvingTypes
    suspicious/noExplicitAny
    suspicious/noExportsInTest
    suspicious/noExtraNonNullAssertion
    suspicious/noFallthroughSwitchClause
    suspicious/noFocusedTests
    suspicious/noFunctionAssign
    suspicious/noGlobalAssign
    suspicious/noGlobalIsFinite
    suspicious/noGlobalIsNan
    suspicious/noImplicitAnyLet
    suspicious/noImportAssign
    suspicious/noImportantInKeyframe
    suspicious/noIrregularWhitespace
    suspicious/noLabelVar
    suspicious/noMisleadingCharacterClass
    suspicious/noMisleadingInstantiator
    suspicious/noMisplacedAssertion
    suspicious/noMisrefactoredShorthandAssign
    suspicious/noOctalEscape
    suspicious/noPrototypeBuiltins
    suspicious/noRedeclare
    suspicious/noSelfCompare
    suspicious/noShadowRestrictedNames
    suspicious/noShorthandPropertyOverrides
    suspicious/noSkippedTests
    suspicious/noSparseArray
    suspicious/noSuspiciousSemicolonInJsx
    suspicious/noTemplateCurlyInString
    suspicious/noThenProperty
    suspicious/noUnsafeDeclarationMerging
    suspicious/noUnsafeNegation
    suspicious/noVar
    suspicious/noWith
    suspicious/useAdjacentOverloadSignatures
    suspicious/useDefaultSwitchClauseLast
    suspicious/useErrorMessage
    suspicious/useGetterReturn
    suspicious/useGoogleFontDisplay
    suspicious/useGuardForIn
    suspicious/useIsArray
    suspicious/useNamespaceKeyword
    suspicious/useNumberToFixedDigitsArgument
    suspicious/useStrictMode
```

### Rule name

lint/nursery/noUselessUndefined

### Playground link

https://biomejs.dev/playground/?lintRules=all&code=KAApACAAPQA%2BACAAdQBuAGQAZQBmAGkAbgBlAGQA&language=ts

### Expected result

unicorn/no-useless-undefined does not flag this.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by vladimir-ivanov** [comment]
I do not see checkArrowFunctionBody option in no_useless_undefined? no options on that rule as yet :-(. 

However there is an interesting conflict of rules:

<img width="581" alt="Image" src="https://github.com/user-attachments/assets/6aab1ff1-0e21-4b5d-bb0f-39d8b2da0839" />

**Comment by siketyan** [comment]
We don't have one yet, but I think we can ignore undefined in an arrow function body by default? May need some discussion

**Comment by vladimir-ivanov** [comment]
<img width="1321" alt="Image" src="https://github.com/user-attachments/assets/3273e41f-489b-4a40-8090-dce078e95ef9" />

**Comment by vladimir-ivanov** [comment]
guess we leave it as is and down to users to deal with it - e.g. turn off one or the other?

Alternatively we should introduce an option on either of the two rules?

@atheck what do you think please?

**Comment by atheck** [comment]
First, thank you for your efforts.

You could have something like this:

```ts
(): string | undefined => undefined
```

In this case only the noUselessUndefined rule would kick in.

To fix this one could write:

```ts
(): string | undefined => {}
```

But then the noEmptyBlockStatements rule would flag this.

In my opinion ignoring `undefined` in arrow functions would be the best option, because in a "normal" function the rule flags if you write `return undefined`. This can be fixed by just writing `return`. In an arrow function you cannot do this. I guess that is why the unicorn eslint plugin introduced that option I meantioned.

**Comment by vladimir-ivanov** [comment]
so your suggestion @atheck is to add `checkArrowFunctionBody` option thanks

**Comment by dyc3** [comment]
I don't feel like we've clearly established the use case that requires adding this option. Is it just so we can let users write empty functions and not conflict with `noEmptyBlockStatements`?

Also regarding return in an arrow function, you should be able to do this:
```
() => { return; }
```

**Comment by Conaclos** [comment]
I think we could make an exception for `() => undefined`. In other cases of the rules the action can simply remove `undefined`, while here it is not the case, we have to turn it into `{}`. And to avoid triggering `noEmptyBlock` this should even be `{ return; }`, `{ ; }`, or `{ /* undefined value */ }` that look hacky.

**Comment by atheck** [comment]
@Conaclos I agree.

**Comment by Conaclos** [comment]
Actually I even wonder if we should not always allow `return undefined`. This is a bit a stretch, however I find that `retrun;` and `return undefined;` convey different meaning: the first indicates an early return in a function that doesn't return any value (`fn(): void`), while the second one indicates a function that explicitly return the `undefined` value (`fn(): something | undefined`).
For example cases like this looks totally fine to me:

```js
export function literalVal(literal: Literal): LiteralVal {
    switch (literal.type) {
        case "bigint":
            return BigInt(literal.val)
        case "number":
        case "string":
            return literal.val
        case "false":
            return false
        case "null":
            return null
        case "true":
            return true
        case "undefined":
            return undefined
    }
}
```

However, this could make the rule very different from its unicorn source.

Also, `useGetterReturn` doesn't accept `return;`, only `return <something>`, including `return undefined` that got flagged by `noUselessUndefined`.

**Comment by atheck** [comment]
Yes, I also agree to that. So, noUselessUndefined would only check parameters I guess.

**Comment by vladimir-ivanov** [comment]
will unasign myself from this
my personal approach on these Is to use use lodash (rxjs) etc -> noop function for such scenarios, not a fan of () => {return;} or () => {;}
Possibly better to leave it without auto fix? and user decides how to deal with it?
